### ISSUE TITLE
Dead code not detected after second assignment and try

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#1900]
```php
$a = [1, 2, 3];
echo($a[0]);
$a = [4, 5, 6];

try {
  // something
} catch (\Throwable $t) {
  // something else
}
```

Expected: second assignment to `$a` is unused
Actual: no issue

https://psalm.dev/r/757378bf44
